platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationContextImpl.java

```diff
@@ -0,0 +1,278 @@
+        this.report = new ExportMigrationReportImpl(report, migratable);
+        this.zos = zos;
+    }
+
+    private static <T> T validateNotNull(T t, String msg) {
+        Validate.notNull(t, msg);
+        return t;
+    }
+
+    @Override
+    public ExportMigrationReportImpl getReport() {
+        return report;
+    }
+
+    @Override
+    public Optional<ExportMigrationEntry> getSystemPropertyReferencedEntry(String name,
+            BiPredicate<MigrationReport, String> validator) {
+        Validate.notNull(name, "invalid null system property name");
+        Validate.notNull(validator, "invalid null validator");
+        final ExportMigrationSystemPropertyReferencedEntryImpl me = systemProperties.get(name);
+
+        if (me != null) {
+            return Optional.of(me);
+        }
+        final String val = System.getProperty(name);
+
+        if (!validator.test(report, val)) {
+            return Optional.empty();
+        } else if (val == null) {
+            report.record(new MigrationException(Messages.EXPORT_SYSTEM_PROPERTY_NOT_DEFINED_ERROR,
+                    name));
+            return Optional.empty();
+        } else if (val.isEmpty()) {
+            report.record(new MigrationException(Messages.EXPORT_SYSTEM_PROPERTY_IS_EMPTY_ERROR,
+                    name));
+            return Optional.empty();
+        }
+        final ExportMigrationSystemPropertyReferencedEntryImpl sprop =
+                new ExportMigrationSystemPropertyReferencedEntryImpl(this, name, val);
+
+        systemProperties.put(name, sprop);
+        return Optional.of(sprop);
+    }
+
+    @Override
+    public ExportMigrationEntry getEntry(Path path) {
+        final ExportMigrationEntryImpl e = new ExportMigrationEntryImpl(this, path);
+
+        return entries.computeIfAbsent(e.getPath(), p -> e);
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path) {
+        final ExportMigrationEntryImpl entry = new ExportMigrationEntryImpl(this, path);
+
+        if (!isDirectory(entry)) {
+            return Stream.empty();
+        }
+        return FileUtils.listFiles(entry.getFile(),
+                TrueFileFilter.INSTANCE,
+                TrueFileFilter.INSTANCE)
+                .stream()
+                .map(File::toPath)
+                .map(this::getEntry);
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path, PathMatcher filter) {
+        final ExportMigrationEntryImpl entry = new ExportMigrationEntryImpl(this, path);
+
+        Validate.notNull(filter, "invalid null filter");
+        if (!isDirectory(entry)) {
+            return Stream.empty();
+        }
+        return FileUtils.listFiles(entry.getFile(),
+                TrueFileFilter.INSTANCE,
+                TrueFileFilter.INSTANCE)
+                .stream()
+                .map(File::toPath)
+                .map(p -> new ExportMigrationEntryImpl(this, p))
+                .filter(e -> filter.matches(e.getPath()))
```

Comment:
❓ Should we filter _before_ creating the `ExportMigrationEntryImpl`?

Reply:
The path has to be cleaned and relativized first.